### PR TITLE
[jsonnet] add support for topology spread for multi zone ingesters

### PR DESCRIPTION
**Pull request by cstyan**
adds jsonnet config for topology spread support (note that I've defaulted it's usage to true) of multi zone ingesters

also refactored the multi zone config fields for better organization/shorter names

Signed-off-by: Callum Styan <callumstyan@gmail.com>

### CODE REVIEW

File: production/ksonnet/loki/multi-zone.libsonnet [hunk 1]
```diff
@@ -17,29 +17,38 @@
...
-    multi_zone_ingester_max_unavailable: std.max(1, std.floor($._config.multi_zone_ingester_replicas / 9)),
-    multi_zone_default_ingester_zone: false,
-    multi_zone_ingester_exclude_default: false,
+    multi_zone_ingesters: {
+      enabled: true,
+      migration_enabled: false,
+      replication_write_path_enabled: true,
+      replication_read_path_enabled: true,
+      num_replicas: 0,
+      max_unavailable: std.max(1, std.floor($._config.multi_zone_ingester.num_replicas / 9)),
```
Comment (dannykopping) [inline production/ksonnet/loki/multi-zone.libsonnet:26]: I think this should be:

```suggestion
      max_unavailable: std.max(1, std.floor($._config.multi_zone.num_replicas / 9)),
```

File: production/ksonnet/loki/multi-zone.libsonnet [hunk 2]
```diff
@@ -17,29 +17,38 @@
...
 
   // Zone-aware replication.
-  distributor_args+:: if !($._config.multi_zone_ingester_enabled && $._config.multi_zone_ingester_replication_write_path_enabled) then {} else {
+  distributor_args+:: if !($._config.multi_zone_ingester.enabled && $._config.multi_zone_ingester.replication_write_path_enabled) then {} else {
     'distributor.zone-awareness-enabled': 'true',
   },
 
   ruler_args+:: (
-    if !($._config.multi_zone_ingester_enabled && $._config.multi_zone_ingester_replication_write_path_enabled) then {} else {
+    if !($._config.multi_zone_ingesterenabled && $._config.multi_zone_ingester.replication_write_path_enabled) then {} else {
```
Comment (dannykopping) [inline production/ksonnet/loki/multi-zone.libsonnet:45]:
```suggestion
    if !($._config.multi_zone.enabled && $._config.multi_zone.replication_write_path_enabled) then {} else {
```

**Review by dannykopping — COMMENTED** [review]
Thanks for jumping on this @cstyan!

Stopped the review though because it seems this refactor is either incomplete or untested; spotted a few issues.

Can you also please add a note to the upgrade guide, since folks using the previous config will need to update it.

**Comment by MasslessParticle** [comment]
Closing for inactivity

**Pull request closed without merging**